tests/ParserTest.php

```diff
@@ -40,6 +40,66 @@ public function provideShortcodes()
+            array('[sc a="x y" b="x" c=""  ]', 'sc', array('a' => 'x y', 'b' => 'x', 'c' => ''), null),
+            array('[sc a="a \"\" b"  ]', 'sc', array('a' => 'a \"\" b'), null),
+            array('[sc/  ]', 'sc', array(), null),
+            array('[sc    /  ]', 'sc', array(), null),
+            array('[sc arg=val cmp="a b"/  ]', 'sc', array('arg' => 'val', 'cmp' => 'a b'), null),
+            array('[sc x y   /  ]', 'sc', array('x' => null, 'y' => null), null),
+            array('[sc x="\ "   /  ]', 'sc', array('x' => '\ '), null),
+
+            //check for whitespace surrounding equals (start data set 43)
+            array('[sc arg  =  val]', 'sc', array('arg' => 'val'), null),
+            array('[sc novalue arg  =  "complex value"]', 'sc', array('novalue' => null, 'arg' => 'complex value'), null),
+            array('[sc x  =  "ąćęłńóśżź ĄĆĘŁŃÓŚŻŹ"]', 'sc', array('x' => 'ąćęłńóśżź ĄĆĘŁŃÓŚŻŹ'), null),
+            array('[sc x  =  "multi'."\n".'line"]', 'sc', array('x' => 'multi'."\n".'line'), null),
+            array('[sc noval x  =  "val" y]content[/sc]', 'sc', array('noval'=> null, 'x' => 'val', 'y' => null), 'content'),
+            array('[sc x  =  "{..}"]', 'sc', array('x' => '{..}'), null),
+            array('[sc a  =  "x y" b  =  "x" c=  ""]', 'sc', array('a' => 'x y', 'b' => 'x', 'c' => ''), null),
+            array('[sc a  =  "a \"\" b"]', 'sc', array('a' => 'a \"\" b'), null),
+            array('[sc arg  =  val cmp  =  "a b"/]', 'sc', array('arg' => 'val', 'cmp' => 'a b'), null),
+            array('[sc x  =  "\ "   /]', 'sc', array('x' => '\ '), null),
+
+            //check for spaces everywhere (start data set 53)
+            array('[  sc  ]', 'sc', array(), null),
+            array('[sc arg  =  val  ]', 'sc', array('arg' => 'val'), null),
+            array('[sc   novalue   arg  =  "complex value"  ]', 'sc', array('novalue' => null, 'arg' => 'complex value'), null),
+            array('[sc x  =  "ąćęłńóśżź ĄĆĘŁŃÓŚŻŹ"  ]', 'sc', array('x' => 'ąćęłńóśżź ĄĆĘŁŃÓŚŻŹ'), null),
+            array('[sc x  =  "multi'."\n".'line"  ]', 'sc', array('x' => 'multi'."\n".'line'), null),
+            array('[sc noval x  =  "val" y  ]content[  /sc  ]', 'sc', array('noval'=> null, 'x' => 'val', 'y' => null), 'content'),
+            array('[sc x  =  "{..}"  ]', 'sc', array('x' => '{..}'), null),
+            array('[sc a  =  "x y" b  =  "x" c  =  ""  ]', 'sc', array('a' => 'x y', 'b' => 'x', 'c' => ''), null),
+            array('[sc a  =  "a \"\" b"  ]', 'sc', array('a' => 'a \"\" b'), null),
+            array('[sc/  ]', 'sc', array(), null),
+            array('[sc    /  ]', 'sc', array(), null),
+            array('[sc arg  =  val cmp  =  "a b"/  ]', 'sc', array('arg' => 'val', 'cmp' => 'a b'), null),
+            array('[sc x y   /  ]', 'sc', array('x' => null, 'y' => null), null),
+            array('[sc x  =  "\ "   /  ]', 'sc', array('x' => '\ '), null),
```

Comment:
All these tests are not necessary and don't provide real value, please add a few examples of those variants of new structure at the end of current `dataProvider`. This is four times more tests to pass which test for the exact same thing.